### PR TITLE
chore(capabilities): define native query connection capabilities

### PR DESCRIPTION
**Pull request by DayTF**
### General

- [ ] Write an explicit title for the Pull Request, following [Conventional Commits specification](https://www.conventionalcommits.org)
- [ ] Test manually the implemented changes
- [ ] Validate the code quality (indentation, syntax, style, simplicity, readability)

### Security

- [ ] Consider the security impact of the changes made

### CODE REVIEW

**Comment by realSpok** [comment]
How is the dev supposed to declare that his datasource supports native queries in this PR ? 🙏

File: packages/datasource-customizer/src/decorators/composite-datasource.ts [hunk 1]
```diff
@@ -62,4 +75,18 @@ export default class CompositeDatasource<T extends Collection = Collection>
 
     throw new Error(`Chart '${name}' is not defined in the dataSource.`);
   }
+
+  async executeNativeQuery(
```
Comment (realSpok) [inline packages/datasource-customizer/src/decorators/composite-datasource.ts:79]: Is there a way to test this at the moment ?

Comment (DayTF) [reply packages/datasource-customizer/src/decorators/composite-datasource.ts:79]: Haven't tested it yet as it is bound to change, I simply made the closest implementation I had in mind for it

File: packages/datasource-customizer/src/decorators/composite-datasource.ts [hunk 2]
```diff
@@ -62,4 +75,18 @@ export default class CompositeDatasource<T extends Collection = Collection>
...
+    contextVariables = {},
+  ): Promise<unknown> {
+    if (!this.nativeQueryConnections[connectionName]) {
+      throw new Error(`Unknown connection name '${connectionName}'`);
+    }
+
+    return this.dataSources
+      .find(datasource => Boolean(datasource.nativeQueryConnections[connectionName]))
+      .executeNativeQuery(connectionName, query, contextVariables);
+  }
```
Comment (realSpok) [inline packages/datasource-customizer/src/decorators/composite-datasource.ts:91]: why not this ? 
```suggestion
    return this.nativeQueryConnections[connectionName]
      .executeNativeQuery(connectionName, query, contextVariables);
  }
```

**Review by realSpok — APPROVED** [review]
- ✅  PR title 
- ✅  Destination branch
- ✅  PR linked to the clickup task

- ℹ️  Code review -> small question
  - ✅  Use pure functions when possible
  - ✅  Performance concerns
  - ✅  Security concerns
  - ✅  The PR tackle one subject only
  
- ✅ Automatic tests
  - ✅  Unit tests
  - ⤬  Integration tests
  
- ✅ Manual tests
  -  ℹ️  Test the functionality. Did not the query execution
  -  ⤬   Test error cases

File: packages/datasource-customizer/src/decorators/composite-datasource.ts (hunk 2 above)
Comment (DayTF) [reply packages/datasource-customizer/src/decorators/composite-datasource.ts:91]: the datasource has the implementation of the executeNativeQuery

**Pull request merged**